sdk/search/azure-search-index/azure/search/index/_index_operation.py

```diff
@@ -0,0 +1,78 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import TYPE_CHECKING
+
+from ._generated.models import IndexAction
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from typing import List
+
+
+__all__ = ("IndexOperationBatch",)
+
+
+class IndexOperationBatch(object):
+    """Represent a batch of upate operations for documents in an Azure
+    Search index.
+
+    Index operations are performed in the order in which they are added
+    to the batch.
+
+    """
+
+    def __init__(self):
+        # type: () -> None
+        self._actions = []  # type: List[IndexAction]
+
+    def __repr__(self):
+        # type: () -> str
+        return "<IndexOperationBatch [{} actions]>".format(len(self.actions))[:1024]
+
+    def add_upload_documents(self, *documents):
```

Comment:
I'm wondering if using a variadic argument here is the best choice. If we assume that the source of the parameters is data driven (e.g. not provided in code, but rather available to the caller as a list/iterable), then having to unpack the parameter value is a bit annoying. And anecdotally, I've also seen users who don't know how to do this.

Thoughts?

Reply:
My actuall preference to write the few gorpy lines of code to make all three versions work:
```
foo(a)          # single item
foo([a, b, c])  # single list
foo(a, b, c)    # args
```
But that's harder to document. Between what's there now, and a single list arg, I don't have a strong pref

Reply:
My own anecdotal personal usability feedback; I find myself almost always appreciating `str.format`'s `*args` parameter (because I *generally* pass in a predefined set of parameter values) whereas I get annoyed with `asyncio.gather` because often I have exploded out an unknown set of tasks to gather. I feel like this api falls in the second bucket.

I do like your gorpy suggestion, though, up until it changes the return type (does it . At which point I have to squint harder to convince myself). For other "batch" APIs that take one or more items (e.g. text analytics), we have gone with a list for among other things, that very reason.

Reply:
I know very little Python, but I do need to stress that indexing a single document at a time is highly inefficient, which is why it's a batch-oriented API. If we want to guide customers towards the pit of success, we should make that as inconvenient to do as possible.

Reply:
@brjohnstmsft, understood (in this case, we are talking about building the batch, so that should not be a concern.)

As a general principle, we do indeed try to make sure that doing the right thing is not less convenient than doing the wrong thing.

Reply:
@johanste agree 100% about consistent return type regardless. I'll plan on supporting all three cases. 

@brjohnstmsft Johan is correct this is about APIs for building batches (methods on a "Batch" object). That said, I did follow the Java API and also add convenience methods on the client for deleting, etc. and those will happily construct batches of 1. Should those be removed? Alternatively, perhaps those convenience methods should only accept an actual list/sequence, so that they are not *too* convenient.

Reply:
Actually the convenience functions are already pluralized, and already accept only explicit lists. I will leave those as-is.